### PR TITLE
Month and year  custom date picker (fix solved)

### PR DESCRIPTION
**Pull request by ferazog**
Add upper limit validation.

### CODE REVIEW

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt [hunk 1]
```diff
@@ -122,18 +123,17 @@ class EditTextMonthYearField(
                                 SimpleDateFormat(showDateFormat, Locale.US).parse(receiver.text.toString())
                         }
 
-                        val datePicker = MonthYearPickerDialog().apply {
+                        MonthYearPickerDialog().apply {
                             year = calendar.get(Calendar.YEAR)
                             month = calendar.get(Calendar.MONTH)
+                            setListener(this@EditTextMonthYearField)
+                            upperLimit = this@EditTextMonthYearField.upperLimit
+                            show(
```
Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: do not use show here, commit allowing state lost this fragment

Example

```
   supportFragmentManager.beginTransaction()
                .add(ErrorDialog.newInstance(), ErrorDialog::class.java.name)
                .commitAllowingStateLoss()
```

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt [hunk 2]
```diff
@@ -162,4 +162,48 @@ class EditTextMonthYearField(
...
+        editText?.text?.substring(
+            MONTH_YEAR_MASK_YEAR_INITIAL_INDEX + 1,
+            MONTH_YEAR_MASK_YEAR_FINAL_INDEX + 1
+        )?.toInt() ?: ZERO
+
+
+    override fun isValid(): ValidationResult {
+        val result = super.isValid()
+        if (result.isValid) {
+            upperLimit?.let {
```
Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: move this to a method, and use early return

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt [hunk 3]
```diff
@@ -162,4 +162,48 @@ class EditTextMonthYearField(
...
 
+    fun getMonth(): Int {
+        val month = editText?.text?.substring(
+            MONTH_YEAR_MASK_MONTH_INITIAL_INDEX,
+            MONTH_YEAR_MASK_MONTH_FINAL_INDEX
+        )?.toInt() ?: ZERO
+        return if (month == ZERO) {
+            ZERO
+        } else {
+            month - 1
```
Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: what does mean this `1`

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt [hunk 4]
```diff
@@ -162,4 +162,48 @@ class EditTextMonthYearField(
...
+        }
+    }
+
+    fun getYear() =
+        editText?.text?.substring(
+            MONTH_YEAR_MASK_YEAR_INITIAL_INDEX + 1,
```
Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: what does mean this `1`

File: app/src/main/java/co/condorlabs/customcomponents/Exceptions.kt [hunk 5]
```diff
@@ -12,4 +12,4 @@ class ViewHolderNotFoundForPositionException(private val position: Int) :
 class ArgumentsNotFoundException: NullPointerException()
 class ArgumentNotFoundException(private val argument: String): NullPointerException("The argument $argument need to be specified")
 class LoadingFragmentListGreaterThatLimitException: IndexOutOfBoundsException("List should only contains 4 elements")
-class NoElementsOnAdapterException : NullPointerException()
+class MonthYearPickerDialogNotFoundException : java.lang.NullPointerException()
```
Comment (diriarte-condor) [inline app/src/main/java/co/condorlabs/customcomponents/Exceptions.kt]: the exception is already imported

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt [hunk 6]
```diff
@@ -163,4 +160,66 @@ class EditTextMonthYearField(
...
+            ZERO
+        } else {
+            month - HUMAN_READABLE_MONTH_INDEX
+        }
+    }
+
+    fun getYear() =
+        editText?.text?.toString()?.let {
+            if(it.isNotEmpty()) {
+                it.replace(SLASH, EMPTY)?.substring(
```
Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: validate that the result is digits only before parse it to int

Comment (ogallon) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: remove `?`

File: app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt [hunk 7]
```diff
@@ -163,4 +160,66 @@ class EditTextMonthYearField(
...
+            if(it.isNotEmpty()) {
+                it.substring(
+                    MONTH_YEAR_MASK_MONTH_INITIAL_INDEX,
+                    MONTH_YEAR_MASK_MONTH_FINAL_INDEX
+                )?.toInt()
+            } else {
+                ZERO
+            }
+        } ?: ZERO
+        return if (month == ZERO) {
```
Comment (diriarte-condor) [inline app/src/main/java/co/condorlabs/customcomponents/customedittext/EditTextMonthYearField.kt]: ZERO should be a meaningful constant

File: test/src/androidTest/java/co/condorlabs/customcomponents/test/util/hooks.kt [hunk 8]
```diff
@@ -57,3 +68,92 @@ fun isTextInLines(lines: Int): TypeSafeMatcher<View> {
...
+                textViewBounds[0] = Point(tvLocation.left, tvLocation.centerY())
+                textViewBounds[1] = Point(tvLocation.centerX(), tvLocation.top)
+                textViewBounds[2] = Point(tvLocation.right, tvLocation.centerY())
+                textViewBounds[3] = Point(tvLocation.centerX(), tvLocation.bottom)
+
+                for (location in DRAWABLE_START_INDEX..DRAWABLE_END_INDEX)
+                    if (drawables[location] != null) {
+                        val bounds = drawables[location].bounds
+                        textViewBounds[location]?.offset(
+                            bounds.width() / 2,
```
Comment (diriarte-condor) [inline test/src/androidTest/java/co/condorlabs/customcomponents/test/util/hooks.kt]: move to a constant

**Pull request merged**